vertical-pod-autoscaler/pkg/recommender/input/history/history_provider.go

```diff
@@ -57,6 +66,7 @@ type PrometheusHistoryProviderConfig struct {
 	CadvisorMetricsJobName                           string
 	Namespace                                        string
 	PrometheusBasicAuthTransport
+	Headers map[string]string
```

Comment:
Could you add unit tests to test the path when Headers are provided in history_provider_test.go?

Reply:
Added please review :)